lib/logstash/inputs/elastic_serverless_forwarder.rb

```diff
@@ -0,0 +1,171 @@
+# encoding: utf-8
+require "logstash/inputs/base"
+require "logstash/namespace"
+
+require "logstash/plugin_mixins/plugin_factory_support"
+
+require 'logstash/inputs/http'
+require 'logstash/codecs/json_lines'
+
+class LogStash::Inputs::ElasticServerlessForwarder < LogStash::Inputs::Base
+  include LogStash::PluginMixins::PluginFactorySupport
+
+  config_name "elastic_serverless_forwarder"
+
+  # no-codec
+  config :codec, obsolete: 'The elastic_serverless_forwarder input does not have an externally-configurable codec'
+
+  # bind address
+  config :host, :validate => :string, :default => "0.0.0.0"
+  config :port, :validate => :number, :required => true
+
+  # optional http basic auth
+  config :auth_basic_username,         :validate => :string
+  config :auth_basic_password,         :validate => :password
+
+  # ssl-config
+  config :ssl,                         :validate => :boolean, :default => true
+
+  # ssl-identity
+  config :ssl_certificate,             :validate => :path
+  config :ssl_key,                     :validate => :path
+  config :ssl_key_passphrase,          :validate => :password
+
+  # ssl-trust
+  config :ssl_certificate_authorities, :validate => :path, :list => true
+  config :ssl_client_authentication,   :validate => %w(none optional required), default: 'none'
+
+  # ssl-expert-mode
+  config :ssl_cipher_suites,           :validate => :string,  :list => true
+  config :ssl_supported_protocols,     :validate => :string,  :list => true
+  config :ssl_handshake_timeout,       :validate => :number
+
+  # we present the ES-like ssl_certificate_authorities, but our
+  # internal http input plugin uses ssl_verify_mode to describe
+  # the same behaviour.
+  SSL_CLIENT_AUTHENTICATION_TO_VERIFY_MODE_MAP = {
+    'none'     => 'none',
+    'optional' => 'peer',
+    'required' => 'force_peer',
+  }.each_value(&:freeze).freeze # deep freeze
+  private_constant :SSL_CLIENT_AUTHENTICATION_TO_VERIFY_MODE_MAP
+
+
+  def initialize(*a)
+    super
+    @internal_http = plugin_factory.input('http').new(inner_http_input_options)
+  end
+
+  def register
+    logger.debug("registering inner HTTP input plugin")
+    @internal_http.register
+    logger.debug("registering inner HTTP input plugin")
```

Comment:
are these log messages supposed to be identical?

Reply:
_NOPE_.

```suggestion
    logger.debug("inner HTTP plugin has been registered")
```